server/server_test.go

```diff
@@ -559,6 +559,43 @@ var _ = Describe("Server package test", func() {
 					testURL("PUT", url, memberTokenID, testResource, http.StatusUnauthorized)
 				})
 			})
+
+			It("should not create resource in different tenant than relation", func() {
+				networkID := "networkred"
+				network := map[string]interface{}{
+					"id":        networkID,
+					"name":      "Networkred",
+					"tenant_id": memberTenantID,
+				}
+				testURL("POST", networkPluralURL, memberTokenID, network, http.StatusCreated)
+				server := map[string]interface{}{
+					"name":       "Server Red",
+					"network_id": networkID,
+					"status":     "ACTIVE",
+					"tenant_id":  powerUserTenantID,
+				}
+
+				testURL("POST", serverPluralURL, memberTokenID, server, http.StatusBadRequest)
```

Comment:
I'm not sure about that, since member should not be able to use other tenants id. Also does this fail for the same reason as the next statement?

Reply:
`since member should not be able to use other tenants id` what do you mean? user might be able to use resource from other tenant if it's "public"